src/implementation/Server/GRPCServer/GRPCServerImpl.ts

```diff
@@ -449,6 +457,75 @@ export default class GRPCServerImpl implements IAppCallbackServer {
     return callback(null, res);
   }
 
+  async onBulkTopicEventAlpha1(
+    call: grpc.ServerUnaryCall<TopicEventBulkRequest, TopicEventBulkResponse>,
+    callback: grpc.sendUnaryData<TopicEventBulkResponse>,
+  ): Promise<void> {
+    const req = call.request;
+    const pubsubName = req.getPubsubName();
+    const topic = req.getTopic();
+
+    // Route is unique to pubsub and topic and has format pubsub--topic--route so we strip it since else we can't find the route
+    const route = this.generatePubSubSubscriptionTopicRouteName(req.getPath().replace(`${pubsubName}--${topic}--`, ""));
+
+    if (
+      !this.pubSubSubscriptions[pubsubName] ||
+      !this.pubSubSubscriptions[pubsubName][topic] ||
+      !this.pubSubSubscriptions[pubsubName][topic].routes[route]
+    ) {
+      this.logger.warn(
+        `The topic '${topic}' is not being subscribed to on PubSub '${pubsubName}' for route '${route}'.`,
+      );
+      return;
+    }
+
+    const resArr: TopicEventBulkResponseEntry[] = [];
+    const entries = req.getEntriesList();
+    for (const ind in entries) {
+      const event = entries[ind];
+      let data: any;
+      if (event.hasBytes()) {
+        data = Buffer.from(event.getBytes()).toString();
+      } else if (event.hasCloudEvent()) {
+        const cloudEvent = event.getCloudEvent();
+        if (cloudEvent instanceof TopicEventCERequest) {
+          data = Buffer.from(cloudEvent.getData()).toString();
+        }
+      }
+      let dataParsed: any;
```

Comment:
Should be merged with PR #453 to abstract the extraction logic

Reply:
Done